Dockerfile

```diff
@@ -1,7 +1,38 @@
-# base image for running tests
-# - preinstalled packages for test speed
+# Base image for running tests.
+#
+# We are using docker as way to test scripts desinged to run inside a regular
+# ubuntu VM. We could user lxd or similar instead, but docker runs OOTB in GH
+# actions.So we install ubuntu-server on top of the ubuntu:20.04 image, and
+# this gets us much closer to a regular VM image.
+#
+# Importantly, we run system as PID 1, like a regular VM. This allows our
+# scripts to install and run systemd services of their own.
+# 
+# Note: this means we need to run this image with some specific options:
+# 
+#   --cap-add SYS_ADMIN \
+#   --tmpfs /tmp --tmpfs /run --tmpfs /run/lock \
+#   -v /sys/fs/cgroup:/sys/fs/cgroup:ro \
+#
+# We also pre-install our packages, so that tests run faster.
 FROM ubuntu:20.04
-ENV DEBIAN_FRONTEND="noninteractive"
-RUN apt-get update && apt-get upgrade -y
+
+ENV container docker
```

Comment:
TIL that `ENV key val` (as opposed to `ENV key=val`) is legal docker syntax. It looks like its deprecated though:
https://docs.docker.com/engine/reference/builder/#env

Reply:
fixed, thanks